### DISCUSSION TITLE
Add possibility to add/delete columns

### DISCUSSION THREAD

**Discussion by Pamwamba** [nick-keller/react-datasheet-grid#237] [Feature requests]
Hi, you did a great work with this datasheet-grid.

It would be great to manage the possibility to add/delete column directly in the component, as the "add rows" feature. I didn't find similar subject so I post it.

When I tried to make custom header to manage it, it caused some issues that I still not understand (https://github.com/nick-keller/react-datasheet-grid/issues/235)

Currently, I'm doing it by adding an external component above the grid, it works, but I have some css issues (width offset between my header and the grid).

This also could be great to edit column names and types maybe with a right clic in the header that could display these feature.

Let me know if this feature can be intersting and added in the future.

**Comment by nick-keller** [comment]
Hey @Pamwamba ,
This component is controlled, meaning it cannot change the columns it receives, it only receives them.
It is up to the parent to add / remove columns as needed.
